src/proto/beats.proto

```diff
@@ -10,22 +11,30 @@ enum Source {
   USER = 2;
 }
 
-message Beat {
-  optional int32 frame_position = 1;
-  optional bool enabled = 2 [ default = true ];
-  optional Source source = 3 [ default = ANALYZER ];
+enum Type {
+  BEAT = 0;
+  BAR = 1;
+  PHRASE = 2;
+}
+
+message TimeSignature {
+  required int32 beats_per_bar = 1 [ default = 4];
+  required int32 note_value = 2 [ default = 4 ];
 }
 
 message Bpm {
   optional double bpm = 1;
   optional Source source = 2 [ default = ANALYZER ];
 }
 
-message BeatMap {
-  repeated Beat beat = 1;
+message Beat {
+  optional double frame_position = 1;
+  optional bool enabled = 2 [ default = true ];
```

Comment:
Why does this field need to be optional?

Reply:
The `enabled` field will be removed in #2844